packages/phone-number-privacy/combiner/src/salt-generation/get-threshold-salt.ts

```diff
@@ -104,6 +108,21 @@ async function requestSignatures(request: Request) {
   return { successCount, signatures, majorityErrorCode }
 }
 
+function requestSigatureWithTimeout(
+  service: SignerService,
+  request: Request
+): Promise<FetchResponse | boolean> {
+  return Promise.race([requestSigature(service, request), abort()])
+}
+
+async function abort(): Promise<boolean> {
```

Comment:
It looks like `node-fetch` has an option for specifying a timeout via an extension:
https://www.npmjs.com/package/node-fetch#options

Reply:
ya i saw that too, but not as clean of an example as what you posted so it has seemed overly complicated.

Reply:
kk updated to use that -- it actually made the tests much easier